src/site/content/en/mini-apps/mini-app-programming-way/index.md

```diff
@@ -0,0 +1,74 @@
+---
+layout: post
+title: Programming the mini app way
+authors:
+  - thomassteiner
+date:  2021-02-05
+# updated: 2021-01-25
+description: |
+  This chapter introduces the way of programming the mini app way.
+tags:
+  - mini-apps
+---
+
+## What has worked well for mini apps
+
+In this chapter, I want to look at lessons I learned from researching mini apps from a web
+developer's point of view, or answer the question what does it mean to develop the mini app way.
+
+## Components
+
+Rather than reinvent the wheel and make developers build yet another implementation of common UI paradigms like tabs,
+accordions, carousels, etc., mini apps just ship with a default selection of components that is extensible in case you need more.
+On the web, there are likewise many options, some of which I have listed in the [chapter on mini app components](/mini-app-components/#web-components).
+In an ideal world, component libraries on the web were built in a way that you could mix them freely.
+In practice, too many times, there is a certain lock-in regarding a design system you need to buy in to when you use a component,
+or the component library is distributed in a way that it is all or nothing, but no individual components can be easily added to a project.
+There are, however, atomic components that you can use in isolation, or libraries like [generic-components](https://github.com/thepassle/generic-components)
+that are unstyled on purpose.
+Finding an using those seems like a good idea.
```

Comment:
```suggestion
```
This is subjective and seems kinda unnecessary. It's safer to just stick to the facts about the differences between the 2 platforms.

Reply:
Very true.